cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -465,5 +465,31 @@ type ComponentConfig interface {
 	Get() interface{}
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResetConfiguration contains a list of elements that is specific "kubeadm reset"-only runtime information.
+type ResetConfiguration struct {
```

Comment:
should we add dry run option?

Reply:
if yes, we can log an issue to add it in init/join configuration too for v1beta4

Reply:
let me check the doc, I remember this was discussed in the doc.

Reply:
ah, yes, 
> This is not configurable in the config file to make it consistent with init /join, i.e. if a user wants to perform a dry run, he/she can only pass it via flag. Open for discussion.

I am +1 to make it configurable, will create one pr to fix it with init/join together.

Reply:
update https://github.com/kubernetes/kubeadm/issues/2890 to add a TODO item to fix it for (reset/init/join/upgrade)

Reply:
@neolit123 update this pr to include the `dryRun` as a config item in the config file, https://github.com/kubernetes/kubernetes/compare/e93b8c83a5f3d871442659e4990be06a213e1ab2..67957a22966531893aca626e44d3eb2d2e420bbe

Could you pls take another look?

Reply:
mark this as resolved since the `dryRun` is now a config item and is has been covered in the testcases, I will propose another  pr for init/join (let's see if we can catch this in 1.28, as the code freeze is only one week away).